source/concurrent/PartiallyGuarded.h

```diff
@@ -0,0 +1,93 @@
+/* PartiallyGuarded.h
+Copyright (c) 2024 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef PARTIALLY_GUARDED_H_
+#define PARTIALLY_GUARDED_H_
+
+#include <list>
+#include <map>
+#include <mutex>
+#include <vector>
+
+
+
+// Containers that guard against some concurrent modifications. Use with caution.
+// For most purposes, this can be treated and passed around like any stl container,
+// but any function modifying it inside a concurrent context MUST receive it as a PartiallyGuarded object
```

Comment:
This sounds like a warning which is likely going to be forgotten by someone, somewhere in the future.
Can we make this PartiallyGuardedList a fully guarded list, by making only the fully implemented features available (and a compile error instead of silently using a dangerous function for non-guarded functions).